qiita_db/analysis.py

```diff
@@ -112,23 +112,53 @@ def create(cls, owner, name, description, parent=None):
             Description of the analysis
         parent : Analysis object, optional
             The analysis this one was forked from
+        from_default : bool, optional
+            If True, use the default analysis to populate selected samples.
+            Default False.
         """
+        queue = "create_analysis"
         conn_handler = SQLConnectionHandler()
+        conn_handler.create_queue(queue)
         # TODO after demo: if exists()
-
-        # insert analysis information into table with "in construction" status
-        sql = ("INSERT INTO qiita.{0} (email, name, description, "
-               "analysis_status_id) VALUES (%s, %s, %s, 1) "
-               "RETURNING analysis_id".format(cls._table))
-        a_id = conn_handler.execute_fetchone(
-            sql, (owner.id, name, description))[0]
+        # Needed since issue #292 exists
+        status_id = conn_handler.execute_fetchone(
+            "SELECT analysis_status_id from qiita.analysis_status WHERE "
+            "status = 'in_construction'")[0]
+        if from_default:
+            # insert analysis and move samples into that new analysis
+            dflt_id = owner.default_analysis
+            sql = """INSERT INTO qiita.{0}
+                    (email, name, description, analysis_status_id)
+                    VALUES (%s, %s, %s, %s)
+                    RETURNING analysis_id""".format(cls._table)
+            conn_handler.add_to_queue(queue, sql, (owner.id, name,
+                                                   description, status_id))
+            # MAGIC NUMBER 3: command selection step
+            # needed so we skip the sample selection step
+            sql = """INSERT INTO qiita.analysis_workflow
```

Comment:
Yeah, I think we are keeping track of everything else in the multiple tables that build up the analysis and the step just limits what can be done on the analysis. I've created a new issue https://github.com/biocore/qiita/issues/1052 and I assigned to beta. Later we can decide to exactly which alpha release it belongs, but I don't think it is either 0.1 or 0.2. Thanks for the feedback!